pkg/ccl/changefeedccl/kvfeed/kv_feed_test.go

```diff
@@ -343,8 +350,56 @@ func TestKVFeed(t *testing.T) {
 				makeTableDesc(42, 1, ts(1), 2, 1),
 				addColumnDropBackfillMutation(makeTableDesc(42, 2, ts(4), 1, 1)),
 			},
-			expEvents: 2,
-			expErrRE:  "schema change ...",
+			expEventsCount: 2,
+			expErrRE:       "schema change ...",
+		},
+		{
+			name:                 "checkpoint events - with quantize",
+			schemaChangeEvents:   changefeedbase.OptSchemaChangeEventClassDefault,
+			schemaChangePolicy:   changefeedbase.OptSchemaChangePolicyBackfill,
+			needsInitialScan:     false,
+			withFrontierQuantize: time.Duration(10),
+			initialHighWater:     ts(2),
+			spans: []roachpb.Span{
+				tableSpan(codec, 42),
+			},
+			events: []kvpb.RangeFeedEvent{
+				checkpointEvent(tableSpan(codec, 42), ts(2)),
+				checkpointEvent(tableSpan(codec, 42), hlc.Timestamp{WallTime: ts(2).WallTime, Logical: 1}),
+				checkpointEvent(tableSpan(codec, 42), hlc.Timestamp{WallTime: ts(3).WallTime + 5, Logical: 1}),
+				checkpointEvent(tableSpan(codec, 42), hlc.Timestamp{WallTime: ts(4).WallTime, Logical: 3}),
+			},
+			expEvents: []kvpb.RangeFeedEvent{
+				checkpointEvent(tableSpan(codec, 42), hlc.Timestamp{WallTime: ts(2).WallTime, Logical: 0}),
+				checkpointEvent(tableSpan(codec, 42), hlc.Timestamp{WallTime: ts(2).WallTime, Logical: 0}),
+				checkpointEvent(tableSpan(codec, 42), hlc.Timestamp{WallTime: ts(3).WallTime, Logical: 0}),
```

Comment:
for my own understanding, why is this 3? seems like it's not quantized to 10s like you specified. shouldnt it be ts(0) or ts(10) and nothing in between?

Reply:
It's related to the weirdness using the helper function ts - https://github.com/cockroachdb/cockroach/blob/cd1cba87ff145928bd18967d6d79b6489d870ae0/pkg/ccl/changefeedccl/kvfeed/kv_feed_test.go#L53-L55. I refactored the test to avoid using this helper function.